### PR TITLE
add completion api support to nvidia inference provider

### PR DESCRIPTION
**Pull request by mattf**
# What does this PR do?

add the completion api to the nvidia inference provider

## Test Plan

while running the meta/llama-3.1-8b-instruct NIM from https://build.nvidia.com/meta/llama-3_1-8b-instruct?snippet_tab=Docker

```
➜ pytest -s -v --providers inference=nvidia llama_stack/providers/tests/inference/ --env NVIDIA_BASE_URL=http://localhost:8000 -k test_completion --inference-model Llama3.1-8B-Instruct
=============================================== test session starts ===============================================
platform linux -- Python 3.10.15, pytest-8.3.3, pluggy-1.5.0 -- /home/matt/.conda/envs/stack/bin/python
cachedir: .pytest_cache
rootdir: /home/matt/Documents/Repositories/meta-llama/llama-stack
configfile: pyproject.toml
plugins: anyio-4.6.2.post1, asyncio-0.24.0, httpx-0.34.0
asyncio: mode=strict, default_loop_scope=None
collected 20 items / 18 deselected / 2 selected                                                                             

llama_stack/providers/tests/inference/test_text_inference.py::TestInference::test_completion[-nvidia] PASSED
llama_stack/providers/tests/inference/test_text_inference.py::TestInference::test_completion_structured_output[-nvidia] SKIPPED

============================= 1 passed, 1 skipped, 18 deselected, 6 warnings in 5.40s =============================
```

the structured output functionality works but the accuracy fails

## Before submitting

- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Ran pre-commit to handle lint / formatting issues.
- [x] Read the [contributor guideline](https://github.com/meta-llama/llama-stack/blob/main/CONTRIBUTING.md),
      Pull Request section?
- [ ] Updated relevant documentation.
- [x] Wrote necessary unit or integration tests.

### CODE REVIEW

**Comment by mattf** [comment]
@ashwinb here's the requested completion implementation for the inference=nvidia provider

File: llama_stack/providers/remote/inference/nvidia/openai_utils.py [hunk 1]
```diff
@@ -31,17 +30,21 @@
     Choice as OpenAIChoice,
     ChoiceLogprobs as OpenAIChoiceLogprobs,  # same as chat_completion_chunk ChoiceLogprobs
 )
-
 from openai.types.chat.chat_completion_message_tool_call_param import (
     Function as OpenAIFunction,
 )
+from openai.types.completion import Completion as OpenAICompletion
```
Comment (raghotham) [inline llama_stack/providers/remote/inference/nvidia/openai_utils.py:36]: Is there a reason to not update/use llama_stack/providers/utils/inference/openai_compat.py instead?

Comment (mattf) [reply llama_stack/providers/remote/inference/nvidia/openai_utils.py:36]: i would like to find a way to blend these together in the future.

despite the file/func names, at the moment neither is truly openai compat.

when i reviewed the openai_compat.py it appeared to have some implementation specific quirks that i assumed are valid for ollama and tgi, but don't hold for nim (or openai). for instance, the eot/eom/python_tag are not returned from the server, and the finish_reason options are different. likewise, there are some nim specific quirks in this implementation, e.g. nvext params, tools are streamed as a single response message, and response_format isn't a thing for openai /v1/completions.

**Review by ashwinb — APPROVED** [review]
looks good to me. there's a merge conflict with the test file.

**Pull request merged**